### PR TITLE
c++17 standard

### PR DESCRIPTION
**Pull request by CusiniM**
Related to PR: https://github.com/GEOS-DEV/GEOS/pull/2496

### CODE REVIEW

File: cmake/CMakeBasics.cmake [hunk 1]
```diff
@@ -12,7 +12,7 @@ option( ENABLE_TOTALVIEW_OUTPUT "" OFF )
 set( LVARRAY_BUILD_OBJ_LIBS OFF CACHE BOOL "" )
 
 
-# if( NOT BLT_CXX_STD STREQUAL c++14 )
+# if( NOT BLT_CXX_STD STREQUAL c++17 )
```
Comment (wrtobin) [inline cmake/CMakeBasics.cmake]: I suppressed this on Frontier when I was swapping between standards to debug very Frontier / cce/hipcc specific issues. 

I think we should probably reactivate it to ensure that all our users / downstream host-configs appropriately update to C++17 when this is merged.

Comment (corbett5) [reply cmake/CMakeBasics.cmake]: I don't know. I would say until we actually start using c++17 features in LvArray (which may be as soon as this is merged IDK), I would leave it at the minimum requirement. I had a branch for my own use and this is what I wound up doing

```
if( CMAKE_CXX_STANDARD IN_LIST "98; 11" )
    MESSAGE(FATAL_ERROR "LvArray requires at least c++14")
endif()
```

Comment (CusiniM) [reply cmake/CMakeBasics.cmake]: yeah, I have just changed it locally. I ll push it later.

File: host-configs/LLNL/lassen-cuda-11-base.cmake [hunk 2]
```diff
@@ -0,0 +1,15 @@
+include(${CMAKE_CURRENT_LIST_DIR}/lassen-base.cmake)
+
+# Cuda options
+set(ENABLE_CUDA ON CACHE BOOL "")
+set(CUDA_TOOLKIT_ROOT_DIR /usr/tce/packages/cuda/cuda-11.2.0 CACHE STRING "")
+set(CMAKE_CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER} CACHE STRING "")
+set(CMAKE_CUDA_COMPILER ${CUDA_TOOLKIT_ROOT_DIR}/bin/nvcc CACHE STRING "")
+set(CUDA_ARCHITECTURES sm_70 CACHE STRING "")
+set(CMAKE_CUDA_STANDARD 14 CACHE STRING "")
```
Comment (rrsettgast) [inline host-configs/LLNL/lassen-cuda-11-base.cmake]:
```suggestion
set(CMAKE_CUDA_STANDARD 17 CACHE STRING "")
```

File: host-configs/LLNL/lassen-clang-10.cmake
Comment (rrsettgast) [inline host-configs/LLNL/lassen-clang-10.cmake:1]: keep the cuda version in the filename

File: host-configs/LLNL/lassen-clang-13.cmake
Comment (rrsettgast) [inline host-configs/LLNL/lassen-clang-13.cmake:1]: keep cuda version in filename

File: host-configs/LLNL/lassen-clang-upstream.cmake
Comment (rrsettgast) [inline host-configs/LLNL/lassen-clang-upstream.cmake:1]: I think we just ditch this host compiler

File: host-configs/LLNL/lassen-cuda-11-base.cmake [hunk 3]
```diff
@@ -0,0 +1,15 @@
+include(${CMAKE_CURRENT_LIST_DIR}/lassen-base.cmake)
+
+# Cuda options
+set(ENABLE_CUDA ON CACHE BOOL "")
+set(CUDA_TOOLKIT_ROOT_DIR /usr/tce/packages/cuda/cuda-11.2.0 CACHE STRING "")
```
Comment (rrsettgast) [inline host-configs/LLNL/lassen-cuda-11-base.cmake]:
```suggestion
set(CUDA_TOOLKIT_ROOT_DIR /usr/tce/packages/cuda/cuda-11.6.1 CACHE STRING "")
```

Comment (CusiniM) [reply host-configs/LLNL/lassen-cuda-11-base.cmake]: do we have 11.6 now? I ll update it on the docker image too then.

Comment (wrtobin) [reply host-configs/LLNL/lassen-cuda-11-base.cmake]: Current lassen cuda module list:
image

Comment (CusiniM) [reply host-configs/LLNL/lassen-cuda-11-base.cmake]: so should we use the latest cuda11 ?

Comment (wrtobin) [reply host-configs/LLNL/lassen-cuda-11-base.cmake]: Looking at the release notes they did some work in 11.7 and 11.8 to reduce both host and device memory footprints, so unless we have a compelling reason not to, I would say go for 11.8 yeah.

**Review by wrtobin — APPROVED** [review]

**Pull request merged**